src/azureFunction/azureFunctionUtils.ts

```diff
@@ -0,0 +1,214 @@
+			continue;
+		} else if (defaultLine.trimStart() === constants.defaultBindingResult) { // Result change
+			replacedValueLines.push(defaultLine.replace(constants.defaultBindingResult, constants.sqlBindingResult));
+		} else {
+			// Normal lines to be included
+			replacedValueLines.push(defaultLine);
+		}
+	}
+	defaultBindedFunctionText = replacedValueLines.join(os.EOL);
+	fs.writeFileSync(filePath, defaultBindedFunctionText, 'utf-8');
+}
+
+/**
+ * Gets the azure function project for the user to choose from a list of projects files
+ * If only one project is found that project is used to add the binding to
+ * if no project is found, user is informed there needs to be a C# Azure Functions project
+ * @returns the selected project file path
+ */
+export async function getAzureFunctionProject(): Promise<string | undefined> {
+	let selectedProjectFile: string | undefined = '';
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return selectedProjectFile;
+	} else {
+		const projectFiles = await getAzureFunctionProjectFiles();
+		if (projectFiles !== undefined) {
+			if (projectFiles.length > 1) {
+				// select project to add azure function to
+				selectedProjectFile = (await vscode.window.showQuickPick(projectFiles, {
+					canPickMany: false,
+					title: constants.selectProject,
+					ignoreFocusOut: true
+				}));
+				return selectedProjectFile;
+			} else if (projectFiles.length === 1) {
+				// only one azure function project found
+				return projectFiles[0];
+			}
+		}
+		return undefined;
+	}
+}
+
+/**
+ * Gets the azure function project files based on the host file found in the same folder
+ * @returns the azure function project files paths
+ */
+export async function getAzureFunctionProjectFiles(): Promise<string[] | undefined> {
+	let projFiles: string[] = [];
+	const hostFiles = await getHostFiles();
+	for (let host of hostFiles) {
+		let projectFile = await vscode.workspace.findFiles('*.csproj', path.dirname(host));
+		projectFile.filter(file => path.dirname(file.fsPath) === path.dirname(host) ? projFiles.push(file?.fsPath) : projFiles);
+	}
+	return projFiles.length > 0 ? projFiles : undefined;
+}
+
+/**
+ * Gets the host files from the workspace
+ * @returns the host file paths
+ */
+export async function getHostFiles(): Promise<string[] | undefined> {
+	const hostUris = await vscode.workspace.findFiles('**/host.json');
+	let hostFiles = [];
+	for (let hostUri of hostUris) {
+		hostFiles.push(hostUri.fsPath);
+	}
+	return hostFiles.length > 0 ? hostFiles : undefined;
+}
+
+/**
+ * Gets the local.settings.json file path
+ * @param projectFile path of the azure function project
+ * @returns the local.settings.json file path
+ */
+export async function getSettingsFile(projectFile: string): Promise<string | undefined> {
+	let settingsFile = path.join(path.dirname(projectFile), 'local.settings.json');
+	return settingsFile.length > 0 ? settingsFile : undefined;
```

Comment:
This check doesn't seem necessary - it shouldn't be possible to have an empty string here